tests/bwc/test_upgrade.py

```diff
@@ -531,5 +520,56 @@ def test_snapshot_compatibility(self):
                 cluster.stop()
                 prev_version = version
                 num_snapshot += 1
-            shutil.rmtree(cluster_settings['path.data'], ignore_errors=True)
+            shutil.rmtree(path_data, ignore_errors=True)
         self._process_on_stop()
+
+class SnapshotHeterogeneousNodesCompatibilityTest(SnapshotCompatibilityTest):
+
+    SUPPORTED_VERSION_TRANSITIONS = ['4.0.x'.replace('x', str(i)) for i in range(0, 9)]
```

Comment:
I'd keep it simple and nail it to `4.0.8` plus `4.0` and remove the `.x` replacement + randomization logic.